lib/SILGen/SILGenDestructor.cpp

```diff
@@ -373,8 +373,11 @@ void SILGenFunction::emitIsolatingDestructor(DestructorDecl *dd) {
 
     // Get deinitOnExecutor
     FuncDecl *swiftDeinitOnExecutorDecl = SGM.getDeinitOnExecutor();
-    assert(swiftDeinitOnExecutorDecl &&
-           "Failed to find swift_task_deinitOnExecutor function decl");
+    if (!swiftDeinitOnExecutorDecl) {
+      llvm::report_fatal_error(
+          "Failed to find swift_task_deinitOnExecutor function decl! "
+          "This is likely due to an outdated/incompatible SDK.");
```

Comment:
Hmm I understood your comment that we do want it to crash, don't we? That seems to be like other code behaves in such situations.

Are you saying we don't want to crash and just return early? I can do that but suspected we'd rather actually fail here as it would not be the right behavior then hm

Reply:
Ideally, the compiler should never crash on any input. We should emit an error and continue normally (generating meaningless code is acceptable).

Reply:
Gotcha, will do 👍

Reply:
Done, actually it seems we can just diagnose in SILGen 
```
if (!swiftDeinitOnExecutorDecl) {
      dd->diagnose(diag::missing_deinit_on_executor_function);
      return;
    }
```